Startup.cs

```diff
@@ -26,19 +27,28 @@ public Startup(IConfiguration configuration, IWebHostEnvironment environment)
         public void ConfigureServices(IServiceCollection services)
         {
             services
-                .AddSingleton<SampleSchema>()
+                .AddSingleton<IDataRetrievalService, DataRetrievalService>()
+                .AddSingleton<IForecastDataRetrievalService, ForecastDataRetrievalService>()
```

Comment:
We should only have one data retrieval service as HTTP Client should only be instantiated once. You may either move the functions in to the `DataRetrivalService` or create a subclass in said service and instantiate it in the constructor of the service.